cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -223,9 +268,10 @@ def get_observations_for_time_range(product_gallery_url, gallery_jwt_token, t1=N
         # format the time fields, from the format request, with +/- 1ms
         t1_minor = parser.parse(t1) - datetime.timedelta(seconds=1)
         t2_plus = parser.parse(t2) + datetime.timedelta(seconds=1)
-        t1_minor_formatted = t1_minor.strftime('%Y-%m-%dT%H:%M:%S')
-        t2_plus_formatted = t2_plus.strftime('%Y-%m-%dT%H:%M:%S')
-        # eg /mmoda-pg/observations/range/2018-12-31T23%3A59%3A59--2021-12-01T00%3A00%3A01
+        # using the format accepted by the product gallery
+        t1_minor_formatted = t1_minor.strftime('%Y-%d-%mT%H:%M:%S')
```

Comment:
omg! We should stick to ISO. Are you sure it uses this weird mix of formats? If yes, it should be exterimnated at the level of drupal, by changing the defaults. It's a source of frequent confusion, and this sort of issues commonly cause issues in astronomy!

Reply:
I mean, if it's not possible to avoid it, we will face troubles, and we will have to keep track very cautiously. It's especially dangerous because of how similar (but seemingly specific) the two formats appear.

Reply:
I have searched and it is affected only in the way the timerange is formatted for this REST call, but I perhaps I will figure out how to customize also this.

I am also worried regarding the potential issues we might face, so I will keep track of it.

In any case, over the product gallery, it is possible to fully customize the format of the time we want to display.

Reply:
For presentation - sure. But two-way translation will be done, in part implicitly.

also you quote this:

> eg /mmoda-pg/observations/range/2018-31-12T23%3A59%3A59--2021-01-12T00%3A00%3A0

does it mean in the urls the time will be the wrong format sometimes?

Reply:
> 
> > eg /mmoda-pg/observations/range/2018-31-12T23%3A59%3A59--2021-01-12T00%3A00%3A0
> 
> does it mean in the urls the time will be the wrong format sometimes?

This is an example of the resulting request directly copied from the views page, the time format will always be this one. What do you mean with wrong?

Reply:
> > > eg /mmoda-pg/observations/range/2018-31-12T23%3A59%3A59--2021-01-12T00%3A00%3A0
> > 
> > 
> > does it mean in the urls the time will be the wrong format sometimes?
> 
> This is an example of the resulting request directly copied from the views page, the time format will always be this one. What do you mean with wrong?

So this sort of URLs will occur, may be copied, and may need to be shared along?

This string: `2018-31-12T23:A5:59` - maybe I could call it not so much wrong, as evil. If it only occurs somewhere in drupal internals - it's ok as long as we do not see if often. If they appear in emails which might be shared -  it's worse.

It's not about matter of taste, we rely on quick look assessment often, and introducing like that an opportunity for mistake.
I have seen multiple times this exact kind of mixed representation causing mistakes, loss of time (of people and observatories) and loss of valuable opportunity. This is no joke. 

I suppose it may be possible to change the default format in drupal. Maybe we can look into that later, if it's not trivial.
If so, let's make an issue.
And discuss it on a meeting.

Reply:
> 
> So this sort of URLs will occur, may be copied, and may need to be shared along?

No, this will only be used internally, no need to share this around.

> 
> I suppose it may be possible to change the default format in drupal. Maybe we can look into that later, if it's not trivial. If so, let's make an issue. And discuss it on a meeting.

This is what I am investigating right now. 

The tricky point here is that the feature to use the timerange as a filter is provided by an external module, and if this does not come with the option to configure the format of the input.

> I suppose it may be possible to change the default format in drupal. Maybe we can look into that later, if it's not trivial.
If so, let's make an issue.

Let's talk about it in a separate issue and decide there.

Reply:
https://github.com/oda-hub/dispatcher-app/issues/329